### PR TITLE
WT-8570 Do not increase oldest ID during recovery

### PR DESCRIPTION
**Pull request by kommiharibabu**
Increasing the oldest ID during the recovery can lead to a
scenario where the eviction threads that operation in recovery
can check the global visibility according to the current oldest ID
and it may go wrong until the recovery rollback to stable is finished.

Also let all the history store records return to the rollback to stable
irrespective of global visibility.

### CODE REVIEW

File: test/suite/test_checkpoint_snapshot05.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            cursor1.set_value(self.valueb)
+            self.assertEqual(cursor1.update(), 0)
+
+        # Create a checkpoint thread
+        done = threading.Event()
+        ckpt = checkpoint_thread(self.conn, done)
+        try:
+            ckpt.start()
+            # Sleep for sometime so that checkpoint starts before committing last transaction.
+            time.sleep(2)
```
Comment (keithbostic) [inline test/suite/test_checkpoint_snapshot05.py:151]: Is there any chance 2 seconds won't be enough on slow machines?

**Review by keithbostic — APPROVED** [review]
lgtm, but I'm not qualified to comment on the heart of the change, the HS flag.

File: test/suite/test_checkpoint_snapshot05.py (hunk 1 above)
Comment (kommiharibabu) [reply test/suite/test_checkpoint_snapshot05.py:151]: There are a couple of other tests with the same 2-second sleep to let the checkpoint start to obtain its snapshot and they didn't fail till now, so I expect that it is enough.

File: src/txn/txn_rollback_to_stable.c [hunk 2]
```diff
@@ -371,7 +371,7 @@ __rollback_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip,
      * outside the constraints of transactions. Therefore, there is no need for snapshot based
      * visibility checks.
      */
-    F_SET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    F_SET(hs_cursor, WT_CURSTD_HS_READ_ALL);
```
Comment (hbokhari) [inline src/txn/txn_rollback_to_stable.c:374]: I assume it's fine to have a value from history store that might have globally visible tombstone?

File: test/suite/test_checkpoint_snapshot05.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                self.assertEqual(v, check_value + str(count + 1))
+            count += 1
+        session.commit_transaction()
+        self.assertEqual(count, nrows)
+
+    def evict_cursor(self, uri, ds, nrows):
+        s = self.conn.open_session()
+        s.begin_transaction()
+        # Configure debug behavior on a cursor to evict the page positioned on when the reset API is used.
+        evict_cursor = s.open_cursor(uri, None, "debug=(release_evict)")
```
Comment (hbokhari) [inline test/suite/test_checkpoint_snapshot05.py:105]: The function name and the variable names are same and that might create some confusion for reader. I will suggest changing function name.

File: src/txn/txn_rollback_to_stable.c (hunk 2 above)
Comment (kommiharibabu) [reply src/txn/txn_rollback_to_stable.c:374]: Yes. All the global visible tombstones according to the recovered checkpoint snapshot are removed by RTS. Whereas the global visibility check in the HS is carried out based on the current global visibility. Anyway, it is protected now by avoiding the increase of the oldest ID. I feel it is better to handle the visibility in the RTS than in the HS.

File: test/suite/test_checkpoint_snapshot05.py (hunk 3 above)
Comment (kommiharibabu) [reply test/suite/test_checkpoint_snapshot05.py:105]: Fixed. Thanks.

**Review by hbokhari — APPROVED** [review]
LGTM!

**Pull request merged**